### PR TITLE
Enhanced error handling and added auto-reboot functionality

### PR DESCRIPTION
**Pull request by HVSharma12**
Changes:

1. Added a comment at the top of the file to explain how to enable auto reboot while running as root and to ensure ansible container is installed as root.

2. Added two new variables: 'ansible_playbook_basename' and 'playbook_path' under 'vars'.

3. Added 'failed_when' conditions to the tasks 'Retrieve image for workload {{ workload.name }}', 'Install tools for workload {{ workload.name }}', and 'Install required packages, if any, for workload {{ workload.name }}' to enhance error handling.

4. Split the 'Handle reboot, if needed' task into two separate tasks: 'Handle manual reboot, if needed' and 'Handle auto reboot, if needed'. The latter includes additional tasks to check if the service unit file exists, create a systemd service unit file if it doesn't exist, enable and start the systemd service, and reboot the system after all tasks.

5. Added a comment to the task 'Enable service to start on reboot for workload {{ workload.name }}' to clarify its purpose.

### CODE REVIEW

File: examples/ansible/roles/alp_reboot/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+markdown
```
Comment (pdostal) [inline examples/ansible/roles/alp_reboot/README.md]:
```suggestion
```

File: examples/ansible/roles/alp_reboot/tasks/end.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  become: true
+  ansible.builtin.file:
+    path: /etc/systemd/system/myplaybook.service
+    state: absent
+
+- name: Remove the myplaybook.service service file
+  become: true
+  ansible.builtin.file:
+    path: "/home"
+    state: touch
```
Comment (pdostal) [inline examples/ansible/roles/alp_reboot/tasks/end.yml]: This makes no sense.

**Review by pdostal — COMMENTED** [review]
This pull request does in my opinion too much at once. It's not even sorted in several commits. Some tasks are negging each other.

File: examples/ansible/roles/alp_reboot/tasks/end.yml (hunk 2 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/tasks/end.yml]: @pdostal, thank you for your review. My apologies for the confusion. That play was indeed intended for testing purposes and was mistakenly included in the commit. I will remove it and push the changes shortly.

Comment (brett060102) [reply examples/ansible/roles/alp_reboot/tasks/end.yml]: @pdostal can you please elaborate on your "too much at once" comment. I see the PR as primarily adding the adding auto-reboot functionality and I think and in general any new functionality would ideally be done in a single PR.

I do see changes in setup_libvirt_host.ym that are more style changes,  like the changes to single lines on some of the "when:" statements and could/should have been done in a different PR. Is that what you meant?

File: examples/ansible/roles/alp_reboot/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/defaults/main.yml]: Not sure need to include "empty" files like this defaults/main.yml or the following handlers/main.yml.

They are not required, and if you are not need/using this files, it is better not to include them.

File: examples/ansible/roles/alp_reboot/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/handlers/main.yml]: See comment about defaults/main.yml above

File: examples/ansible/roles/alp_reboot/meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  # issue_tracker_url: http://example.com/issue/tracker
+
+  # Choose a valid license ID from https://spdx.org - some suggested licenses:
+  # - BSD-3-Clause (default)
+  # - MIT
+  # - GPL-2.0-or-later
+  # - GPL-3.0-only
+  # - Apache-2.0
+  # - CC-BY-4.0
+  license: license (GPL-2.0-or-later, MIT, etc)
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/meta/main.yml]: You should pick an appropriate license value here, which would be Apache-2.0 given that that is the overall license for this repo.

```suggestion
  license: Apache-2.0
```

File: examples/ansible/roles/alp_reboot/meta/main.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+  min_ansible_version: 2.1
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/meta/main.yml]: Given that all of the testing that has been done has been with Ansible 7.x or later I would change this to 7.0. And I'd say that it is unlikely that this code will work with Ansible versions <2.9.

```suggestion
  min_ansible_version: 7.0
```

File: examples/ansible/roles/alp_reboot/README.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+## Role Variables
+
+The main variable used in this role is `ansible_user`, which specifies the user account to enable lingering for and to execute tasks that require elevated privileges. Additionally, `playbook_path`, `ansible_playbook_basename`, and `reboot_mode` are also defined in this role.
+
+## Tasks Included in This Role
+
+Here's an overview of the tasks executed by this role:
+
+1. Enable lingering for the `ansible_user`.
+2. Check if the service unit file `/etc/systemd/system/myplaybook.service` exists. If it does, remove it.
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/README.md]: I would suggest using a service name that is more descriptive and general purpose, e.g. resume-playbook.service.

File: examples/ansible/roles/alp_reboot/README.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+3. Create a systemd service unit file from the template `myplaybook.service.j2` only if it did not exist previously.
+4. Enable and start the `myplaybook` service, only if the service unit file did not previously exist.
+5. Reboot the system if the service unit file did not previously exist.
+6. Remove the `myplaybook.service` service file.
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/README.md]: Same goes for the `myplaybook` references here and throughout the rest of the PR.

File: examples/ansible/roles/alp_reboot/README.md (hunk 7 above)
Comment (rtamalin) [reply examples/ansible/roles/alp_reboot/README.md]: Or better yet `resume-ansible-playbook`

File: examples/ansible/roles/alp_reboot/meta/main.yml [hunk 9]
```diff
@@ -0,0 +1,52 @@
+galaxy_info:
+  author: your name
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/meta/main.yml]: You may want to fill in your details here, and in the following lines.

File: examples/ansible/roles/alp_reboot/meta/main.yml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  # min_ansible_container_version:
+
+  #
+  # Provide a list of supported platforms, and for each platform a list of versions.
+  # If you don't wish to enumerate all versions for a particular platform, use 'all'.
+  # To view available platforms and versions (or releases), visit:
+  # https://galaxy.ansible.com/api/v1/platforms/
+  #
+  # platforms:
+  # - name: Fedora
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/meta/main.yml]: Technically you should be providing a platforms entr

File: examples/ansible/roles/alp_reboot/meta/main.yml [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  # If this a Container Enabled role, provide the minimum Ansible Container version.
+  # min_ansible_container_version:
+
+  #
+  # Provide a list of supported platforms, and for each platform a list of versions.
+  # If you don't wish to enumerate all versions for a particular platform, use 'all'.
+  # To view available platforms and versions (or releases), visit:
+  # https://galaxy.ansible.com/api/v1/platforms/
+  #
+  # platforms:
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/meta/main.yml]: Technically you should be provide a platforms specification indicating that the role is specific to SUSE ALP here, though I'm not sure if ALP is "known" to the Ansible Galaxy registry as a platform yet...

Or if it is more generally applicable then perhaps the role name `alp_reboot` is not appropriate and perhaps the role should be called `resume_after_reboot` or something along those lines.

File: examples/ansible/roles/alp_reboot/tasks/end.yml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+- name: Remove the myplaybook.service service file
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/tasks/end.yml]: See above comments about using a more appropriate service name.

File: examples/ansible/roles/alp_reboot/tasks/start.yml [hunk 13]
```diff
@@ -0,0 +1,41 @@
+- name: Enable lingering for the user
+  ansible.builtin.command:
+    cmd: "loginctl enable-linger {{ ansible_user }}"
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/tasks/start.yml]: Note that this could be considered a potential security risk if left enabled, so you potentially should be disabling it again, if you had to enabled it, after successfully rebooting.

Also it might be a good idea to provide a comment block here explaining why lingering is needed?

File: examples/ansible/roles/alp_reboot/README.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+You can include this role in your Ansible playbook to manage system reboots.
+
+The include_role syntax in the playbook looks like this:
+
+```yaml
+
+- name: Handle auto reboot, if needed
+  include_role:
+    name: alp_reboot
+    tasks_from: start
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/README.md]: Perhaps `setup` rather than `start` may be a better name for the task list?

File: examples/ansible/roles/alp_reboot/README.md [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+    The when: required_pkgs_install.changed and (reboot_mode | default('manual')) == 'auto' part specifies the condition under which the tasks from the included role should be executed. In this case, the alp_reboot role will be included if the required_pkgs_install result has changed (i.e., packages have been installed) and the reboot_mode variable is set to 'auto'.
+
+Similarly, towards the end of the playbook, the alp_reboot role is included again but with tasks_from: end. This refers to the end.yml file within the tasks directory of the alp_reboot role:
+
+```yaml
+
+- name: clean up
+  include_role:
+    name: alp_reboot
+    tasks_from: end
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/README.md]: Similarly maybe `cleanup` rather than `end` may be a better name for the task list?

File: examples/ansible/roles/alp_reboot/README.md [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+Usage
+
+You can include this role in your Ansible playbook to manage system reboots.
+
+The include_role syntax in the playbook looks like this:
+
+```yaml
+
+- name: Handle auto reboot, if needed
+  include_role:
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/README.md]: You probably should be using fully qualified action names here, similar to what is used in the code itself.

File: examples/ansible/roles/alp_reboot/README.md [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+7. Reload the systemd daemon.
+8. Conditionally end the playbook execution.
+
+
+
+Usage
+
+You can include this role in your Ansible playbook to manage system reboots.
+
+The include_role syntax in the playbook looks like this:
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/README.md]: Also consider referring people to the documentation for the the `include_role` action, accessible either via the `ansible-doc` command or via the official Ansible docs website https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_role_module.htmlhttps://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_role_module.html

File: examples/ansible/roles/alp_reboot/tasks/start.yml [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+
+- name: Check if the service unit file exists
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/tasks/start.yml]: What happens if two different playbooks are running on the system, and want to configure resumption after reboot?

With this logic flow, the second playbook to run would remove the first playbook's service definition...

I understand that this shouldn't really be happening, but what if someone configured a playbook to run after a reboot, that in turn needs to trigger a reboot, and then someone else, or even the same user, runs a playbook after the first reboot, but before the second reboot, or before the resumed playbook got to the cleanup phase? In such a case we could "break" things due to the racing actions...

Comment (rtamalin) [reply examples/ansible/roles/alp_reboot/tasks/start.yml]: Note that one way of dealing with this would be to define an appropriate service template, i.e. a service unit with an `@` on the end of the name, and require the use of the role to provide a service instance identifier, and then configure specific instances of that service for the provide instance name.

That would partially mitigate things, and would even allow you to define multiple playbooks to be run after a reboot...

However, there would still be the potential for those playbooks to race against each other as they could be triggered at the same time.

Comment (rtamalin) [reply examples/ansible/roles/alp_reboot/tasks/start.yml]: Another alternative would be to define a persistent generic service definition that checks a particular directory location, e.g. /etc/ansible-resume-playbooks/, for resumption config files, which will be processed in order by the service, which avoid the risk of racing resume playbooks.

However, you would need to ensure that if a resumed playbook itself added an entry, to be resumed after the next reboot, then it should not be resumed until after the next reboot.

File: examples/ansible/roles/alp_reboot/templates/myplaybook.service.j2 [hunk 19]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=My Ansible Playbook
+After=network.target
+
+[Service]
+Type=oneshot
+WorkingDirectory={{ playbook_path }}
+ExecStart=ansible-playbook {{ ansible_playbook_basename }}  -e 'playbook_path={{ playbook_path }}' -e 'ansible_user={{ ansible_user }}' -e 'reboot_mode=auto'
```
Comment (rtamalin) [inline examples/ansible/roles/alp_reboot/templates/myplaybook.service.j2]: Note that the enablement of auto reboot when calling the ansible playbook in this service could lead to "infinite" reboots if the user have a logic bug in the resumed playbook that keeps configuring additional reboot resumptions...

You may want to make the reboot_mode setting that is passed to the resumed playbook something that the user of the role can specify/override if they want to avoid such a problem scenario.

File: examples/ansible/setup_libvirt_host.yml [hunk 20]
```diff
@@ -4,10 +4,14 @@
 # A example playbook showing how the setup an ALP system as a libvirt host.
 #
 
+# use ansible-playbook setup_libvirt_host.yml -e "playbook_path=$(pwd)" -e "ansible_user=$(whoami)" -e "reboot_mode=auto" to enable auto reboot while running as root ensure ansible container is installed as root.
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: This comment is very long and should be reformatted as a multi-line comment:

```suggestion
# Use:
#   ansible-playbook setup_libvirt_host.yml -e "playbook_path=$(pwd)"
#       -e "ansible_user=$(whoami)"
#       -e "reboot_mode=auto"
# to enable auto reboot while running as root to ensure ansible container
# is installed as root.
```

Comment (rtamalin) [reply examples/ansible/setup_libvirt_host.yml]: Also it would probably be better to say that the Ansible container needs to be pre-loaded into the target user's image cache to avoid long delays running ansible commands as that user, if it is not the same user that is running the current playbook.

File: examples/ansible/setup_libvirt_host.yml [hunk 21]
```diff
@@ -4,10 +4,14 @@
...
 
 - name: Setup ALP system as a libvirt host
   hosts: alphost
   vars:
+    ansible_playbook_basename: "setup_libvirt_host.yml"
+    ansible_user: "{{ ansible_user }}"
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml:18]: This defined ansible_user as itself, which should be a value that originally comes from the inventory specifying what alphost is.

I'm not sure you should be doing this is the correct thing to do here, or makes sense.

File: examples/ansible/setup_libvirt_host.yml [hunk 22]
```diff
@@ -4,10 +4,14 @@
 # A example playbook showing how the setup an ALP system as a libvirt host.
 #
 
+# use ansible-playbook setup_libvirt_host.yml -e "playbook_path=$(pwd)" -e "ansible_user=$(whoami)" -e "reboot_mode=auto" to enable auto reboot while running as root ensure ansible container is installed as root.
 
 - name: Setup ALP system as a libvirt host
   hosts: alphost
   vars:
+    ansible_playbook_basename: "setup_libvirt_host.yml"
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml:17]: Facts/variables starting with `ansible_...` are usually reserved for the Ansible tool itself to use, and I don't see this being defined as a "standard" special variable/fact as documented here: https://docs.ansible.com/ansible/latest/reference_appendices/special_variables.html

I too have wished there was a fact/variable identifying the top-level playbook that is being run from inside an ansible task, but I don't believe there is a way to do so.

I would recommend changing this fact name to something more appropriate to the use case, e.g. alpreboot_resume_playbook, and either define it explicitly via set_fact action with an accompanying name: item that indicates why it is being defined, or as a "paramter" to the alp_reboot role, via a vars block when it is being included, to indicate the intended scope of where it is being used...

File: examples/ansible/setup_libvirt_host.yml [hunk 23]
```diff
@@ -25,36 +29,35 @@
...
+      failed_when: retrieve_workload_image is failed
 
     - name: Install tools for workload {{ workload.name }}
       become: true
       ansible.builtin.command: >-
         podman container runlabel {{ workload.labels.install }} {{ workload.image }}
       register: workload_runlabel_install
-      changed_when:
-        - ('already exist' not in workload_runlabel_install.stdout)
+      changed_when: "'already exist' not in workload_runlabel_install.stdout"
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: This is an unnecessary formatting/whitespace change to the existing code, and necessitated the additional double quoting of the condition which was already validly using parentheses wrapping to avoid complaints about a condition entry starting with a quote character.

File: examples/ansible/setup_libvirt_host.yml [hunk 24]
```diff
@@ -25,36 +29,35 @@
...
 
     - name: Install required packages, if any, for workload {{ workload.name }}
       become: true
       ansible.builtin.package:
         name: "{{ workload.required_pkgs }}"
         state: present
       register: required_pkgs_install
-      when:
-        - ((workload.required_pkgs | default([])) | length) > 0
+      when: ((workload.required_pkgs | default([])) | length) > 0
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: This is an unnecessary formatting/whitespace change to the existing; changes to existing code should be minimised when proposing new changes, and new code should be written to conform to the same coding style/pattern as the original code.

If there is a desire to change the coding style of the code that should be handled as a separate PR that makes no functional changes to the code.

File: examples/ansible/setup_libvirt_host.yml [hunk 25]
```diff
@@ -25,36 +29,35 @@
...
+      failed_when: required_pkgs_install is failed
+
 
-    - name: Handle reboot, if needed
-      when:
-        - required_pkgs_install is changed
+    - name: Handle manual reboot, if needed
+      when: required_pkgs_install.changed and (reboot_mode | default('manual')) == 'manual'
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: Note that the `*when` condition list syntax avoids the need to explicitly use the `and` keyword when adding additional conditions to an exiting condition list, as the conditions specified in the list are implicitly "anded" together.

This leads to easier to read code and eliminates the need for changes to existing lines in such cases.

```suggestion
      when:
        - required_pkgs_install.changed
        - (reboot_mode | default('manual')) == 'manual'
```

File: examples/ansible/setup_libvirt_host.yml [hunk 26]
```diff
@@ -67,55 +70,49 @@
...
+    - name: Handle auto reboot, if needed # Check if automatic reboot is required
+      include_role:
+        name: alp_reboot
+        tasks_from: start
+      when: required_pkgs_install.changed and (reboot_mode | default('manual')) == 'auto'
+
     - name: Setup service for workload {{ workload.name }}
-      when:
-        - workload.service is defined
+      when: workload.service is defined
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: Unnecessary formatting/whitespace change.

File: examples/ansible/setup_libvirt_host.yml [hunk 27]
```diff
@@ -67,55 +70,49 @@
...
 
-        - name: Enable service to start on reboot for workload {{ workload.name }}
+        - name: Enable service to start on reboot for workload {{ workload.name }}   # Enable the service to start on reboot
           become: true
           ansible.builtin.systemd_service:
             name: "{{ workload.service.name }}"
             enabled: true
-          when:
-            - workload_service_start is success
+          when: workload_service_start is success
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: Unnecessary formatting/whitespace change.

File: examples/ansible/setup_libvirt_host.yml [hunk 28]
```diff
@@ -67,55 +70,49 @@
...
 
         - name: Wait for service to be ready for workload {{ workload.name }}
-          when:
-            - workload.service.wait_for is defined
+          when: workload.service.wait_for is defined
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: Unnecessary formatting/whitespace change.

File: examples/ansible/setup_libvirt_host.yml [hunk 29]
```diff
@@ -67,55 +70,49 @@
...
           block:
-
             - name: Check service path for workload {{ workload.name }}
               ansible.builtin.wait_for:
                 path: "{{ workload.service.wait_for.path }}"
-              when:
-                - workload.service.wait_for.path is defined
+              when: workload.service.wait_for.path is defined
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: Unnecessary formatting/whitespace change.

File: examples/ansible/setup_libvirt_host.yml [hunk 30]
```diff
@@ -67,55 +70,49 @@
...
             - name: Check service port for workload {{ workload.name }}
               ansible.builtin.wait_for:
                 port: "{{ workload.service.wait_for.port }}"
-              when:
-                - workload.service.wait_for.port is defined
-
-#
-# Verify basic operation of the Libvirt service
-#
+              when: workload.service.wait_for.port is defined
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: Unnecessary formatting/whitespace change.

File: examples/ansible/setup_libvirt_host.yml [hunk 31]
```diff
@@ -125,5 +122,10 @@
     - name: Report running Libvirt VMs, if any
       ansible.builtin.debug:
         msg: "Running VMs: {{ virt_list_vms.list_vms | join(', ') }}"
-      when:
-        - ((virt_list_vms.list_vms | default([])) | length) > 0
+      when: ((virt_list_vms.list_vms | default([])) | length) > 0
```
Comment (rtamalin) [inline examples/ansible/setup_libvirt_host.yml]: Unnecessary formatting/whitespace change.

**Comment by rtamalin** [comment]
@HVSharma12 As a general comment it is a good idea to avoid making unnecessary changes to existing code, such as reformatting existing valid code, as doing so unnecessarily bloats the changes in the PR, and provides no actual benefit.

It is better to ensure that new code conforms to the existing code foramtting/style, to keep things consistent, when proposing new changes.

If there is a desire to change the formatting/style of existing code then that should be done via a PR that only makes those changes, without making any functional changes.

**Comment by rtamalin** [comment]
@HVSharma12 As regards the automated reboot mechanism, I believe that what you have here is a good start, but probably needs a little more thought to handle problematic edge cases, such as multiple racing playbook runs, and possible security implications of registering an arbitrary playbook to be run automatically on reboot.

File: examples/ansible/roles/alp_reboot/tests/test.yml [hunk 32]
```diff
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - alp_reboot
```
Comment (pdostal) [inline examples/ansible/roles/alp_reboot/tests/test.yml]: How will this be executed?

**Comment by pdostal** [comment]
Do I understand correctly that the purpose of `alp_reboot` role is to enable other ansible playbooks to be running from the host against the host and not be interrupted by the host being rebooted? Isn't it just a workaround?

File: examples/ansible/roles/alp_reboot/defaults/main.yml (hunk 3 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/defaults/main.yml]: Removed unused ansible role files

File: examples/ansible/roles/alp_reboot/handlers/main.yml (hunk 4 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/handlers/main.yml]: Removed unused ansible role file

File: examples/ansible/roles/alp_reboot/tests/test.yml (hunk 32 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/tests/test.yml]: Removed unused ansible role files

File: examples/ansible/roles/alp_reboot/README.md (hunk 14 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/README.md:39]: Renamed the role tasks and updated the same in the include_role task.

File: examples/ansible/roles/alp_reboot/README.md (hunk 15 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/README.md:57]: Renamed the role tasks and updated the same in the include_role task

Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/README.md]: Renamed the role tasks and updated the same in the include_role task, will be updating the readme shortly.

File: examples/ansible/roles/alp_reboot/README.md (hunk 14 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/README.md]: Renamed the role tasks and updated the same in the include_role task, will be updating the readme shortly.

File: examples/ansible/roles/alp_reboot/tasks/start.yml (hunk 13 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/tasks/start.yml]: added a comment block explaining why lingering is needed and a cleanup task to disable lingering.

File: examples/ansible/roles/alp_reboot/README.md (hunk 7 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/README.md]: renamed myplaybook service to resume-ansible-playbook in the role templates,cleanup.yml and setup.yml, will update in the readme shortly.

File: examples/ansible/roles/alp_reboot/README.md (hunk 8 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/README.md]: renamed myplaybook service to resume-ansible-playbook in the role templates,cleanup.yml and setup.yml, will update in the readme shortly.

File: examples/ansible/roles/alp_reboot/meta/main.yml (hunk 11 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/meta/main.yml]: renamed role from alp_reboot to resume_after_reboot and updated the include_role task in the playbook with the new role name.

File: examples/ansible/roles/alp_reboot/tasks/end.yml (hunk 12 above)
Comment (HVSharma12) [reply examples/ansible/roles/alp_reboot/tasks/end.yml]: updated the service name

File: examples/ansible/setup_libvirt_host.yml (hunk 31 above)
Comment (HVSharma12) [reply examples/ansible/setup_libvirt_host.yml]: followed the same coding style/pattern as the original code

**Comment by HVSharma12** [comment]
Adopting a new approach due to edge case considerations. Closing this PR; will open a new one for better clarity.
